### PR TITLE
Adds a better/safer method to generate embeds

### PR DESCRIPTION
**Pull request by tarzzz**

### CODE REVIEW

**Comment by tarzzz** [comment]
@scjody Please review..

File: plotlyhtmlexporter/preprocessors.py [hunk 1]
```diff
@@ -17,13 +32,23 @@ def sanitize_code_outputs(self, outputs):
         """
         outputs = super(PlotlySanitizeHTML, self).sanitize_code_outputs(outputs)
 
-        # Make `text/vnd.plotly.v1+html` content (which is trusted)
-        # available as `text/html`.
+        context = {'unique_div_id': uuid.uuid4()}
+
+        # Parse `text/vnd.plotly.v1+json` content and make it available as 'text/html'
```
Comment (scjody) [inline plotlyhtmlexporter/preprocessors.py]: `text` or `application`?

Comment (tarzzz) [reply plotlyhtmlexporter/preprocessors.py]: Technically, it should be `application/html`, but 

  - in ipython notebooks, all the outputs are processed into `text/html` by default. 
  - The display-priority list of the parent class (HTMLExporter) does not mention `application/html` : https://github.com/jupyter/nbconvert/blob/master/nbconvert/exporters/html.py#L50
so would require additional changes, to put `application/html`s display-priority above `text/html` ..

Comment (tarzzz) [reply plotlyhtmlexporter/preprocessors.py]: Update: their core template: https://github.com/jupyter/nbconvert/blob/master/nbconvert/templates/skeleton/display_priority.tpl
Does not recognize `application/html` as a mime-type ..

Comment (scjody) [reply plotlyhtmlexporter/preprocessors.py]: Why does this comment refer to `text/vnd.plotly.v1+json` when the rest of the code uses `application/vnd.plotly.v1+json`?

Comment (tarzzz) [reply plotlyhtmlexporter/preprocessors.py]: Yep, sorry that was a typo.. will update it.

**Comment by scjody** [comment]
@tarzzz 

**How much of this is cut and pasted from streambed? Can you put a note in both places as a reminder to update one when updating the other?**

**Comment by tarzzz** [comment]
@scjody Yes, I am  going to do it today. I had planned on taking up another minor (and related) issue on streambed, so I will add a note alongwith it..

**Comment by scjody** [comment]
It would have been better to do the plotlyhtmlexporter part in this PR, but I look forward to seeing both PRs today.

**Pull request merged**